aws-ts-static-website/index.ts

```diff
@@ -145,14 +126,73 @@ if (config.certificateArn === undefined) {
     certificateArn = certificateValidation.certificateArn;
 }
 
+/**
+ * Only provision a Lambda@Edge fuction (and related resources) if languageRedirect is 'true' from configuration.
+ * Once provisioned test with the following curl commands
+ * - curl -L -H 'accept-language: en' $(pulumi stack output targetDomainEndpoint)
+ * - curl -L -H 'accept-language: es' $(pulumi stack output targetDomainEndpoint)
+ */
+let languageCacheBehavior = [];
+if (config.languageRedirect) {
+    const languageRedirectRole = new aws.iam.Role("language-redirect", {
+        assumeRolePolicy: JSON.stringify({
+            Version: "2012-10-17",
+            Statement: [
+                {
+                    Action: "sts:AssumeRole",
+                    Principal: {
+                        Service: [
+                            "lambda.amazonaws.com",
+                            "edgelambda.amazonaws.com",
+                        ],
+                    },
+                    Effect: "Allow",
+                    Sid: "",
+                },
+            ],
+        }),
+    });
+
+    const languageRedirectLambda = new aws.lambda.Function("language-redirect", {
+        code: new pulumi.asset.AssetArchive({
+            "index.js": new pulumi.asset.FileAsset("./src/lambda-language-redirect.js"),
+        }),
+        role: languageRedirectRole.arn,
+        handler: "index.handler",
+        publish: true,
+        runtime: aws.lambda.NodeJS8d10Runtime,
+    }, { provider: eastRegion });
```

Comment:
Is this not supported (yet) with `aws.lambda.CallbackFunction`?  That would in principle allow you to write the handler in-line here - which could be quite slick for this sort of use case (very small lambdas generally for Lambda@Edge, and very much part of the infrastructure behavior). 

Cc @cyrusnajmabadi.

Reply:
agreed. this looks like this should be doable using 'aws.lambda.CallbackFunction'.  Ideally it is self explanatory to do so.  If you try it out and it doesn't work, please feel free to ping me on slack to help you through it.

Reply:
Done